server/PokerTableObjects/PokerTable.js

```diff
@@ -0,0 +1,590 @@
+        for(let i = 0; i < 2; i++){ // each player is dealt two cards
+            for(let j = 0; j < this.getNumberOfPlayersAbleToAct(); j++){ // since everyone playing is able to act at the start, this is appropriate to use
+                pos = this.findNextPlayerToDistributeACard(pos); // starting one seat left of the dealer distribute a card
+                this.tableSeats[pos].cards[i] = this.deckCards[this.currentDeckCard];
+                this.currentDeckCard++;
+            }
+        }
+        this.setCommunityCards();
+    }
+
+    findNextPlayerToDistributeACard(pos){ // without incrementing the turn we want to find the next player in the game, used for distributing cards
+        for(let i = 0; i < 6; i++){
+            pos++;
+            if(pos > 5){
+                pos = 0;
+            }
+            if(!this.tableSeats[pos].seatOpen){
+                // found the next player
+                return pos;
+            }
+        }
+    }
+
+    findNextTurn(){ // find the next player who can make a decision
+        // first check if one or less players is able to make decisions
+        if(this.getNumberOfPlayersAbleToAct() < 2  && this.allAblePlayersMadeDecision()){
+            // one or less players remaining, we collect the pots and move straight to the show down
+            console.log('premature show down');
+            this.beginTheShowDown();
+        }
+        else{
+            // we go though all the seats until we find a player
+            for(let i = 0; i < 6; i++){
+                this.seatTurnID++;
+                if(this.seatTurnID > 5){
+                    this.seatTurnID = 0;
+                }
+                if(this.tableSeats[this.seatTurnID].isPlayerSeatAbleToAct()){// found the next player
+                    return;
+                }
+            }
+            console.log('error');
+        }
+    }
+
+    makeDealer(){
+        this.tableSeats[this.seatTurnID].dealer = true;
+        this.dealerSeatID = this.seatTurnID;
+    }
+
+    getNumberOfPlayersAbleToAct(){ // returns the number of players still in play that can make decisions
+        let count = 0;
+        for(let i = 0; i < 6; i++){
+            if(this.tableSeats[i].ableToAct){
+                count++;
+            }
+        }
+        return count;
+    }
+
+    makeBlind(blind){
+        let tableSeat = this.tableSeats[this.seatTurnID];
+        // deduct up to the blind, go all in if they can't make the pay
+        if(tableSeat.chips > blind){
+            tableSeat.bet = blind;
+            tableSeat.chips -= blind;
+        }
+        else{
+            tableSeat.bet = tableSeat.chips;
+            tableSeat.chips = 0;
+            tableSeat.action = 'ALL IN';
+            tableSeat.ableToAct = false;
+        }
+    }
+
+    setplayerTurn(){
+        // setup so the current turn is ready to make a decision
+        this.tableSeats[this.seatTurnID].turn = true;
+    }
+
+    setCommunityCards(){
```

Comment:
Could this function be refactored to use a loop?

Reply:
I'm not sure, the rules of poker state that a card is thrown away (burned) before the next community card reveal. and the order is 3 revealed then 1 revealed then 1 revealed.
and then after a card is set/burned we increment the pointer

so the pattern is:
burn-reveal-increment-reveal-increment-reveal-increment-burn-reveal-increment-burn-reveal-increment

which would be confusing to put as a loop

Reply:
Seems to me that at the very least lines 270 - 275 could be refactored.  Open to discussion though - I just think that this function seems like a bit of a code smell

Reply:
This might be dumb, but does burning a card mean anything in terms of gameplay? As in, is it necessary to do in an online setting without a physical deck of cards?

Reply:
technically no, i guess removing that would simplify it and allow it to become a loop. ill change it